Add descriptions of configurations to the docs (#725)
ACCESS-NRI/access-esm1.6-configs: documentation/docs/pages/configs_experiments/experiments/1pctCO2.md

The 1pctCO2 configuration simulates the climate under a yearly 1% increase in CO2 concentrations, starting from preindustrial conditions.

The configuration matches the [preindustrial+concentrations](/configs_experiments/configurations/preindustrial+concentrations) configuration with changes for prescribing the yearly CO2 increase.

This configuration is used for the [CMIP7 1pctCO2 experiment](https://airtable.com/embed/apphXCUgASIeT6jCz/shrCs1cSWzQRV0v4i/tblbT6XAdQYOCMXu7/viwUXPlXGkKPiFTgB/recdrTGc9OOrRF1rU).

> 

Reviewer: Maybe this preamble section needs an extra sentence to reinforce that the described have been made, the description is for understanding / to assist in other modifications only ?

Author: I've added the following for each configuration:
```
Please note that the settings described below are implemented in the configuration and no further changes are required to run it. The descriprions below are included to aid in understanding of the configuration and to assist with making modifications.
```

#### atmosphere/namelists
```diff
 &CLMCHFCG
- L_CLMCHFCG=.FALSE.,
+ L_CLMCHFCG=.TRUE.,
+ CLIM_FCG_NYEARS(1)= 1,
+ CLIM_FCG_YEARS(1,1)=0,
+ CLIM_FCG_LEVLS(1,1)=4.3189e-04,
+ CLIM_FCG_RATES(1,1)= 1.00000,
+ CLIM_FCG_NYEARS(2:11)=10*0,
 /
```
In the above:

* `L_CLMCHFCG=.TRUE.` enables time varying prescribed greenhouse gas concentrations.
* `CLIM_FCG_NYEARS(1)= 1` indicates that one year of prescribed CO2 concentrations will be provided.
* `CLIM_FCG_YEARS(1,1)=200` indicates that the CO2 concentrations are provided for calendar year 200.
* `CLIM_FCG_LEVLS(1,1)=4.3189e-04` sets the CO2 mass mixing ratio for calendar year 200.
* `CLIM_FCG_RATES(1,1)= 1.00000` tells the model to apply a 1% increase to the CO2 concentration for each year after the last provided prescribed value.

> * `CLIM_FCG_NYEARS(2:11)=10*0` tells the model that time varying values are not being provided for the other greenhouse gasses. Their values are instead taken from the `&RUN_Radiation` section.

Reviewer: is it useful to try and link to code or docs for each options ?

Reviewer 2: The docs won't be publicly available.

Author: Unfortunately there don't seem to be any UM docs which explain this, and the code won't be available to most people

With these settings, the model applies a 1% increase in CO2 concentrations for each year following the calendar year set in `CLIM_FCG_YEARS(1,1)=200`, so that calendar year 201 uses a concentration of `4.3189e-04*1.01=4.362089e-04`, calendar year 202 uses `4.3189e-04*1.01^2=4.4057098e-04` and so on. The year in the namelist has been set to 200, as the initial restart files used by the configuration have a calendar year of 201. It's important for the dates in the initial restart to be one year greater than the value set for `CLIM_FCG_YEARS(1,1)`, otherwise the incorrect increases in CO2 concentrations will be applied.

The values for other greenhouse gas concentrations, volcanic forcings, and the solar constant are indentical to the [piControl](/configs_experiments/configurations/piControl) configuration.

## Configuration scripts
The 1pctCO2 configuration includes a `check_co2_year.py` userscript which is run during the payu setup stage. This checks during the first run that the restart year is one year greater than the value set for `CLIM_FCG_YEARS(1,1)`, and produces an error otherwise. 

!!! warning

>     This check will not work properly if the payu run index is customised with the `payu run -i` option.

Reviewer 2: Do we want to add a snippet from the payu config file to show where this script is specified? 

Do we want to give the path to the script in the configuration, or is the path enforced by payu?

Author: Good idea. I've added in the following snippet which also shows the relative path to the script:
https://github.com/ACCESS-NRI/access-esm1.6-configs/blob/e6d62e94e5fd83812d4124415e4c6c7a7a9cd858/documentation/docs/pages/configs_experiments/configurations/1pctCO2.md?plain=1#L48-L52

!!! warning

    The ACCESS-ESM1.6 1pctCO2 experiment does not have a released and maintained configuration. These notes describe how the CMIP7 experiment was set up and are for reference only. If you would like to run a 1pctCO2 experiment with the latest configuration updates, please create a *help request* on the [ACCESS Hive Forum](https://forum.access-hive.org.au/t/support-faq-frequently-asked-questions/1021).

The 1pctCO2 experiment simulates the climate under a yearly 1% increase in CO2 concentrations, starting from preindustrial conditions.

> The experimental setup matches the [piControl](/configs_experiments/configurations/piControl) configuration with changes for prescribing the yearly CO2 increase and changes to the initial conditions.

Reviewer 2: Should we specify the version of the piControl configuration that it matches? To re-emphasise that the configurations might evolve but not the experiments.

Author: There's a small complication as the 1pctCO2 experiment was run before any release tags existed for the piControl configuration. We could point to the commit it is based on, or still point to the release tag with a comment to clarify that it's not the exact base that was used. What do you think would be best here?

Reviewer 2: I think I prefer the truth, so the commit hash would be best. Can we add our own "non-release" tag otherwise to help here?

Author: Sounds good! I've just added the relevant commit hashes for the esm-flat10, abrupt-4xCO2, and 1pctCO2 pages in 3b655dd

This experiment has been run for the [CMIP7 1pctCO2 experiment](https://airtable.com/embed/apphXCUgASIeT6jCz/shrCs1cSWzQRV0v4i/tblbT6XAdQYOCMXu7/viwUXPlXGkKPiFTgB/recdrTGc9OOrRF1rU).

## Inputs

> The 1pctCO2 experiment uses the same input files as the [piControl](/configs_experiments/configurations/piControl) configuration other than the initial conditions.

Reviewer 2: ```suggestion
The 1pctCO2 experiment uses the same input files as the [piControl](/configs_experiments/configurations/piControl) configuration except for the initial conditions.
```
The "other than" sounds weird when I read, but it could just be me.

### Restart
The 1pctCO2 experiment uses the restart from the end of the 100th year of the piControl experiment.

## Key settings
The yearly 1% increase in atmospheric CO2 concentrations is configured in the `atmosphere/namelists` file. Under the `&CLIMCHFG` section, an initial concentration and year are specified along with a rate of increase to apply for the following years.

The following differences are shown with respect to the [piControl](/configs_experiments/configurations/piControl) configuration:

#### atmosphere/namelists
```diff
 &CLMCHFCG
- L_CLMCHFCG=.FALSE.,
+ L_CLMCHFCG=.TRUE.,
+ CLIM_FCG_NYEARS(1)= 1,
+ CLIM_FCG_YEARS(1,1)=200,
+ CLIM_FCG_LEVLS(1,1)=4.3189e-04,
+ CLIM_FCG_RATES(1,1)= 1.00000,
+ CLIM_FCG_NYEARS(2:11)=10*0,
 /
```
In the above:

* `L_CLMCHFCG=.TRUE.` enables time varying prescribed greenhouse gas concentrations.
* `CLIM_FCG_NYEARS(1)= 1` indicates that one year of prescribed CO2 concentrations will be provided.

> * `CLIM_FCG_YEARS(1,1)=200` indicates that the CO2 concentrations are provided for calendar year 200.

Reviewer 2: Why year 200? Is that 200 years after the start of the 1pctCO2 run or, since we take the restart at year 100 of piControl, is it 100 years after the start of the 1pctCO2 run? What CO2 concentration is used for the first 100 years or so?

Should we talk of "simulation year" or "calendar year"? Considering we don't give a reference calendar, "calendar year" is weird.

* `CLIM_FCG_LEVLS(1,1)=4.3189e-04` sets the CO2 mass mixing ratio for calendar year 200 to equal the value used in the [piControl](/configs_experiments/configurations/piControl) configuration.
* `CLIM_FCG_RATES(1,1)= 1.00000` tells the model to apply a 1% increase to the CO2 concentration for each year after the last provided prescribed value.
* `CLIM_FCG_NYEARS(2:11)=10*0` tells the model that time varying values are not being provided for the other greenhouse gasses. Their values are instead taken from the `&RUN_Radiation` section.

> With these settings, the model applies a 1% increase in CO2 concentrations for each year following the calendar year set in `CLIM_FCG_YEARS(1,1)=200`, so that calendar year 201 uses a concentration of `4.3189e-04*1.01=4.362089e-04`, calendar year 202 uses `4.3189e-04*1.01^2=4.4057098e-04` and so on. The year in the namelist has been set to 200, as the initial restart files used by the configuration have a calendar year of 201. It's important for the dates in the initial restart to be one year greater than the value set for `CLIM_FCG_YEARS(1,1)`, otherwise the incorrect increases in CO2 concentrations will be applied.

Reviewer 2: The explanation on the year needs to be up with the dot point on the "200". And the bit about calendar year 201 in the restart files needs to be in the Restart section that comes above. Because otherwise the reader is lost between the 100th restart and the calendar year 200.

Author: Good point, its not very clear as is. I've tried rearranging based on your suggestion and have changed it to

> * `CLIM_FCG_YEARS(1,1)=200` indicates CO2 mass mixing ratios in the `CLIM_FCG_LEVLS(1,1)` setting are provided for calendar year 200. This has been set to be one less than 201, the calendar year in the initial restart file. It's important for this setting to be exactly one year less than the year in the initial restart file, otherwise incorrect increases in CO2 concentrations will be applied.
>...
>With these settings, the model applies a 1% increase in CO2 concentrations for each year following the calendar year set in `CLIM_FCG_YEARS(1,1)=200`, so that calendar year 201 uses a concentration of `4.3189e-04*1.01=4.362089e-04`, calendar year 202 uses `4.3189e-04*1.01^2=4.4057098e-04` and so on.
